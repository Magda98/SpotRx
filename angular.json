{
	"$schema": "./node_modules/@angular/cli/lib/config/schema.json",
	"version": 1,
	"newProjectRoot": "projects",
	"projects": {
		"spotrx": {
			"projectType": "application",
			"schematics": {
				"@schematics/angular:component": {
					"style": "scss"
				},
				"@schematics/angular:application": {
					"strict": true
				}
			},
			"root": "",
			"sourceRoot": "src",
			"prefix": "app",
			"architect": {
				"build": {
					"builder": "@angular-devkit/build-angular:application",
					"options": {
						"outputPath": "dist/spotrx",
						"index": "src/index.html",
						"browser": "src/main.ts",
						"tsConfig": "tsconfig.app.json",
						"inlineStyleLanguage": "scss",
						"assets": ["src/favicon.ico", "src/assets"],
						"styles": ["src/styles.scss", "src/styles/m3-theme.scss"]
					},
					"configurations": {
						"production": {
							"budgets": [
								{
									"type": "initial",
									"maximumWarning": "1.1mb",
									"maximumError": "2mb"
								},
								{
									"type": "anyComponentStyle",
									"maximumWarning": "2kb",
									"maximumError": "4kb"
								}
							],
							"fileReplacements": [
								{
									"replace": "src/environments/environment.ts",
									"with": "src/environments/environment.prod.ts"
								}
							],
							"outputHashing": "all"
						},
						"development": {
							"optimization": false,
							"extractLicenses": false,
							"sourceMap": true
						}
					},
					"defaultConfiguration": "production"
				},
				"serve": {
					"builder": "@angular-devkit/build-angular:dev-server",
					"configurations": {
						"production": {
							"buildTarget": "spotrx:build:production"
						},
						"development": {
							"buildTarget": "spotrx:build:development"
						}
					},
					"defaultConfiguration": "development"
				},
				"extract-i18n": {
					"builder": "@angular-devkit/build-angular:extract-i18n",
					"options": {
						"buildTarget": "spotrx:build"
					}
				},
				"deploy": {
					"builder": "angular-cli-ghpages:deploy"
				},
				"storybook": {
					"builder": "@storybook/angular:start-storybook",
					"options": {
						"configDir": ".storybook",
						"browserTarget": "spotrx:build",
						"styles": ["src/styles.scss", "src/styles/m3-theme.scss"],
						"compodoc": false,
						"compodocArgs": ["-e", "json", "-d", "."],
						"port": 6006
					}
				},
				"build-storybook": {
					"builder": "@storybook/angular:build-storybook",
					"options": {
						"configDir": ".storybook",
						"browserTarget": "spotrx:build",
						"styles": ["src/styles.scss", "src/styles/m3-theme.scss"],
						"compodoc": false,
						"compodocArgs": ["-e", "json", "-d", "."],
						"outputDir": "storybook-static"
					}
				},
				"lint": {
					"builder": "@angular-eslint/builder:lint",
					"options": {
						"lintFilePatterns": ["src/**/*.ts", "src/**/*.html"]
					}
				}
			}
		}
	},
	"cli": {
		"analytics": false,
		"schematicCollections": ["@angular-eslint/schematics"]
	}
}
