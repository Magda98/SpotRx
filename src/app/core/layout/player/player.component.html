@if(player) {
<div class="player">
  <ng-container>
    @if(playerService.playbackState | async; as position){
    <mat-slider class="song-slider" min="1" [max]="player.duration" step="10">
      <input (dragEnd)="setPosition($event)" [value]="position" matSliderThumb>
    </mat-slider>
    }
    <div class="song-info">
      <img [alt]="player.track_window.current_track.album.name" class="album-image"
        [src]="player.track_window.current_track.album.images[0].url">
      <div class="album-artist">
        <h2 class="name">{{player.track_window.current_track.name}}</h2>
        <span class="album-name">
          {{player.track_window.current_track.album.name}}
        </span>
        <div class="artist-container">
          @for(item of player.track_window.current_track.artists; track item.name){
          <span class="artist">{{item.name}}</span>
          }
        </div>
      </div>
    </div>
    <div class="action-buttons-container">
      <div class="action-buttons">
        <button appButton type="button" aria-label="previous" [mode]="'text'" (click)="playerService.skipPrev()">
          <mat-icon>skip_previous</mat-icon>
        </button>
        <button appButton type="button" aria-label="play" [mode]="'text'" (click)="playerService.tooglePlay()">
          @if(player.paused){
          <mat-icon>play_circle_outline</mat-icon>
          }
          @else{
          <mat-icon>pause_circle_outline</mat-icon>
          }
        </button>
        <button appButton type="button" aria-label="next" [mode]="'text'" (click)="playerService.skipNext();">
          <mat-icon>skip_next</mat-icon>
        </button>
      </div>
    </div>
    <div class="slider-container">
      <button appButton type="button" [mode]="'text'">
        <app-icon [ngClass]="{'primary': player.shuffle }" (click)="toggleShuffle()" class="icon"
          type="shuffle"></app-icon>
      </button>
      <button appButton type="button" [mode]="'text'" (click)="toggleSavedTrack()">
        <app-icon [ngClass]="{'primary': isSavedTrack() === true }" class="icon" type="heart"></app-icon>
      </button>
      <mat-slider class="volume-slider" min="0" max="1" step="0.1">
        <input (dragEnd)="setVolume($event)" value="0.5" matSliderThumb>
      </mat-slider>
    </div>
  </ng-container>
</div>
}