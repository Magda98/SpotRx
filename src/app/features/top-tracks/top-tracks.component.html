<div class="top-tracks">
	<h1 class="title">Top tracks 😎</h1>
	<app-track-list
		class="track-list"
		(getNextPage)="getNextPage($event)"
		[total]="this.topTracksQuery.data()?.total"
		[tracksList]="topTracks()"
		[isLoading]="topTracksQuery.isLoading()">
	</app-track-list>
</div>
