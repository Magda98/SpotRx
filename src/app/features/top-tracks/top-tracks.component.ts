import { Component, computed, inject, signal } from '@angular/core';
import { TrackListComponent } from '../../shared/components/track-list/track-list.component';
import { injectQuery } from '@tanstack/angular-query-experimental';
import { TrackService } from '../../shared/services/track.service';
import { PageEvent } from '@angular/material/paginator';
import { Item, TracksResponse } from '../../shared/interfaces/track';

@Component({
	selector: 'app-top-tracks',
	templateUrl: './top-tracks.component.html',
	styleUrls: ['./top-tracks.component.scss'],
	imports: [TrackListComponent],
})
export class TopTracksComponent {
	trackService = inject(TrackService);
	offset = signal(0);
	topTracksQuery = injectQuery(() => this.trackService.getTopTracks(this.offset()));
	topTracks = computed(() => {
		const topTracksResponse = this.topTracksQuery.data();
		if (!topTracksResponse) return;
		const mappedTracks: Item[] =
			topTracksResponse?.items.map((item) => ({
				track: item,
			})) ?? [];
		const mappedResponse: TracksResponse = {
			...topTracksResponse,
			items: mappedTracks,
		};
		return mappedResponse;
	});
	getNextPage(page: PageEvent) {
		this.offset.set(page.pageSize * page.pageIndex);
	}
}
