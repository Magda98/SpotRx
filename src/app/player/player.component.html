<div class="player">
    <!-- <mat-slider min="1" max="5" step="0.5" value="1.5"></mat-slider> -->
    <button *ngIf="playerState | async as player" class="play" (click)="togglePlay()">
        <mat-icon *ngIf="player.paused">play_circle_outline</mat-icon>
        <mat-icon *ngIf="!player.paused">pause_circle_outline</mat-icon>
        {{player.position}}{{player.paused}}
    </button>

</div>