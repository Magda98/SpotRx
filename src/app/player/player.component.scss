.player {
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 140px;
  background-color: #2c2f31;
  padding-left: 0;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr;
  padding-top: 16px;

  @media screen and (min-width: 980px) {
    grid-template-columns: 1fr 1fr 1fr;
    padding-left: 220px;
  }

  .song-slider {
    position: absolute;
    top: -23px;
    left: 0;
    right: 0;

    @media screen and (min-width: 980px) {
      left: 212px;
    }
  }

  .action-buttons {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-left: 0;
    padding: 0 8px;

    @media screen and (min-width: 980px) {
      margin-left: 25%;
      width: 100%;
      max-width: 250px;
    }
  }

  .song-info {
    display: flex;
    align-items: center;
    padding: 0 8px;

    .album-image {
      width: 60px;
      height: 60px;
      object-fit: cover;
      border-radius: 5px;
      display: block;
      margin-right: 10px;

      @media screen and (min-width: 980px) {
        width: 80px;
        height: 80px;
      }
    }

    .name {
      margin-bottom: 0;
    }

    .artist-container {
      display: flex;
      opacity: 0.6;
      margin-top: 5px;
    }

    .artist {
      margin-right: 5px;
    }

    .album-artist {
      display: flex;
      flex-direction: column;
    }

    .album-name {
      opacity: 0.6;
      font-size: 12px;

      @media screen and (min-width: 980px) {
        font-size: 16px;
      }
    }
  }

  .slider-container {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .volume-slider {
    display: none;

    @media screen and (min-width: 980px) {
      display: block;
      max-width: 150px;
      width: 100%;
    }
  }
}
