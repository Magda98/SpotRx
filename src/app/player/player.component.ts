import { PlayerService } from './../services/player.service';
import { Component, OnDestroy, OnInit } from '@angular/core';
import { MatSliderDragEvent } from '@angular/material/slider';
import {
  EMPTY,
  Subscription,
  distinctUntilChanged,
  finalize,
  map,
  switchMap,
  tap,
} from 'rxjs';
import { TrackService } from '../services/track.service';

@Component({
  selector: 'app-player',
  templateUrl: './player.component.html',
  styleUrls: ['./player.component.scss'],
})
export class PlayerComponent implements OnInit, OnDestroy {
  public playerState = this.playerService.getPlayerState();
  public player?: Spotify.PlaybackState;
  public isSavedTrack?: boolean;
  private subscription = new Subscription();
  private toggleTrackSubscription?: Subscription;
  private isFetching = false;

  constructor(
    public playerService: PlayerService,
    private trackService: TrackService
  ) {}

  ngOnInit(): void {
    this.subscription.add(
      this.playerState
        .pipe(
          tap((player) => {
            this.player = player;
          }),
          map((player) => player.track_window.current_track.id),
          distinctUntilChanged(),
          switchMap((id) => {
            if (id) return this.trackService.checkUserSavedTrack(id);
            return EMPTY;
          })
        )
        .subscribe((data) => {
          const [isSavedTrack] = data;
          this.isSavedTrack = isSavedTrack;
        })
    );
  }

  toggleSavedTrack() {
    if (!this.player?.track_window?.current_track?.id) return;

    if (this.isSavedTrack) {
      this.toggleTrackSubscription?.unsubscribe();
      return (this.toggleTrackSubscription = this.trackService
        .deleteTrack(this.player.track_window.current_track.id)
        .subscribe(() => {
          this.isSavedTrack = false;
        }));
    }

    this.toggleTrackSubscription?.unsubscribe();
    return (this.toggleTrackSubscription = this.trackService
      .saveTrack(this.player.track_window.current_track.id)
      .subscribe(() => {
        this.isSavedTrack = true;
      }));
  }

  setVolume(volumeSlider: MatSliderDragEvent) {
    this.playerService.setVolume(volumeSlider.value);
  }

  setPosition(positionSlider: MatSliderDragEvent) {
    this.playerService.setPosition(positionSlider.value);
  }

  toggleShuffle() {
    if (!this.player) return;
    if (this.isFetching) return;

    this.isFetching = true;
    this.player.shuffle = !this.player.shuffle;
    this.subscription.add(
      this.trackService
        .toggleShuffle(this.player.shuffle)
        .pipe(
          finalize(() => {
            this.isFetching = false;
          })
        )
        .subscribe()
    );
  }

  ngOnDestroy(): void {
    this.subscription.unsubscribe();
    this.toggleTrackSubscription?.unsubscribe();
  }
}
