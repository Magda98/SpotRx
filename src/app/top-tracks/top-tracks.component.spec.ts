import { render } from '@testing-library/angular';
import { TopTracksComponent } from './top-tracks.component';
import {
  QueryClient,
  provideAngularQuery,
} from '@tanstack/angular-query-experimental';
import { PlayerService } from '../services/player.service';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { MockBackendInterceptor } from '../../tests/mock-backend.interceptor';
import { HttpClientTestingModule } from '@angular/common/http/testing';

describe('TopTracksComponent', () => {
  test('should render', async () => {
    const component = await render(TopTracksComponent, {
      imports: [HttpClientTestingModule],
      providers: [
        provideAngularQuery(new QueryClient()),
        PlayerService,
        {
          provide: HTTP_INTERCEPTORS,
          useClass: MockBackendInterceptor,
          multi: true,
        },
      ],
    });
    jest.advanceTimersByTime(300);
    component.detectChanges();
    expect(component.container).toMatchSnapshot();
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });
});
