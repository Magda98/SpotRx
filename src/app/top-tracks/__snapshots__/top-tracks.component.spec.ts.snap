// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTracksComponent should render 1`] = `
<div
  id="root0"
>
  <div
    class="top-tracks"
  >
    <h1
      class="title"
    >
      Top tracks ðŸ˜Ž
    </h1>
    <app-track-list
      class="track-list"
    >
      <div
        class="track-list-container"
      >
        <mat-paginator
          aria-label="Select page"
          class="mat-mdc-paginator paginator"
          role="group"
        >
          <div
            class="mat-mdc-paginator-outer-container"
          >
            <div
              class="mat-mdc-paginator-container"
            >
              <div
                class="mat-mdc-paginator-page-size"
              >
                <div
                  class="mat-mdc-paginator-page-size-label"
                >
                   Items per page: 
                </div>
                
                <div
                  class="mat-mdc-paginator-page-size-value"
                >
                  6
                </div>
                
              </div>
              
              <div
                class="mat-mdc-paginator-range-actions"
              >
                <div
                  aria-live="polite"
                  class="mat-mdc-paginator-range-label"
                >
                   0 of 0 
                </div>
                
                <button
                  aria-label="Previous page"
                  class="mat-mdc-tooltip-trigger mat-mdc-paginator-navigation-previous mdc-icon-button mat-mdc-icon-button mat-mdc-button-disabled _mat-animation-noopable mat-unthemed mat-mdc-button-base mat-mdc-tooltip-disabled"
                  disabled="true"
                  mat-icon-button=""
                  mat-ripple-loader-centered=""
                  mat-ripple-loader-class-name="mat-mdc-button-ripple"
                  mat-ripple-loader-disabled=""
                  mat-ripple-loader-uninitialized=""
                  type="button"
                >
                  <span
                    class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
                  />
                  <svg
                    aria-hidden="true"
                    class="mat-mdc-paginator-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                  <span
                    class="mat-mdc-focus-indicator"
                  />
                  <span
                    class="mat-mdc-button-touch-target"
                  />
                </button>
                
                <button
                  aria-label="Next page"
                  class="mat-mdc-tooltip-trigger mat-mdc-paginator-navigation-next mdc-icon-button mat-mdc-icon-button mat-mdc-button-disabled _mat-animation-noopable mat-unthemed mat-mdc-button-base mat-mdc-tooltip-disabled"
                  disabled="true"
                  mat-icon-button=""
                  mat-ripple-loader-centered=""
                  mat-ripple-loader-class-name="mat-mdc-button-ripple"
                  mat-ripple-loader-disabled=""
                  mat-ripple-loader-uninitialized=""
                  type="button"
                >
                  <span
                    class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
                  />
                  <svg
                    aria-hidden="true"
                    class="mat-mdc-paginator-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <span
                    class="mat-mdc-focus-indicator"
                  />
                  <span
                    class="mat-mdc-button-touch-target"
                  />
                </button>
                
                
              </div>
            </div>
          </div>
        </mat-paginator>
        
        <div
          class="track-list track-list--skeleton"
        >
          <button
            class="track"
            type="button"
          >
            <div
              class="song-info"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 90px; width: 90px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
              <div
                class="album-artist"
              >
                <ngx-skeleton-loader
                  animation="progress-dark"
                  count="1"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 32px; width: 160px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
                <ngx-skeleton-loader
                  animation="progress-dark"
                  class="artist-skeleton"
                  count="2"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
              </div>
            </div>
            <span
              class="time"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 16px; width: 30px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
            </span>
          </button>
          <button
            class="track"
            type="button"
          >
            <div
              class="song-info"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 90px; width: 90px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
              <div
                class="album-artist"
              >
                <ngx-skeleton-loader
                  animation="progress-dark"
                  count="1"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 32px; width: 160px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
                <ngx-skeleton-loader
                  animation="progress-dark"
                  class="artist-skeleton"
                  count="2"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
              </div>
            </div>
            <span
              class="time"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 16px; width: 30px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
            </span>
          </button>
          <button
            class="track"
            type="button"
          >
            <div
              class="song-info"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 90px; width: 90px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
              <div
                class="album-artist"
              >
                <ngx-skeleton-loader
                  animation="progress-dark"
                  count="1"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 32px; width: 160px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
                <ngx-skeleton-loader
                  animation="progress-dark"
                  class="artist-skeleton"
                  count="2"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
              </div>
            </div>
            <span
              class="time"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 16px; width: 30px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
            </span>
          </button>
          <button
            class="track"
            type="button"
          >
            <div
              class="song-info"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 90px; width: 90px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
              <div
                class="album-artist"
              >
                <ngx-skeleton-loader
                  animation="progress-dark"
                  count="1"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 32px; width: 160px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
                <ngx-skeleton-loader
                  animation="progress-dark"
                  class="artist-skeleton"
                  count="2"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
              </div>
            </div>
            <span
              class="time"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 16px; width: 30px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
            </span>
          </button>
          <button
            class="track"
            type="button"
          >
            <div
              class="song-info"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 90px; width: 90px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
              <div
                class="album-artist"
              >
                <ngx-skeleton-loader
                  animation="progress-dark"
                  count="1"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 32px; width: 160px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
                <ngx-skeleton-loader
                  animation="progress-dark"
                  class="artist-skeleton"
                  count="2"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
              </div>
            </div>
            <span
              class="time"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 16px; width: 30px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
            </span>
          </button>
          <button
            class="track"
            type="button"
          >
            <div
              class="song-info"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 90px; width: 90px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
              <div
                class="album-artist"
              >
                <ngx-skeleton-loader
                  animation="progress-dark"
                  count="1"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 32px; width: 160px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
                <ngx-skeleton-loader
                  animation="progress-dark"
                  class="artist-skeleton"
                  count="2"
                >
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  <div
                    aria-busy="true"
                    aria-label="loading"
                    aria-valuemax="100"
                    aria-valuemin="0"
                    aria-valuetext="Loading..."
                    class="skeleton-loader progress-dark"
                    role="progressbar"
                    style="height: 16px; width: 150px; background-color: rgb(44, 47, 49);"
                    tabindex="-1"
                  >
                    
                  </div>
                  
                </ngx-skeleton-loader>
              </div>
            </div>
            <span
              class="time"
            >
              <ngx-skeleton-loader
                animation="progress-dark"
                count="1"
              >
                <div
                  aria-busy="true"
                  aria-label="loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuetext="Loading..."
                  class="skeleton-loader progress-dark"
                  role="progressbar"
                  style="height: 16px; width: 30px; background-color: rgb(44, 47, 49);"
                  tabindex="-1"
                >
                  
                </div>
                
              </ngx-skeleton-loader>
            </span>
          </button>
          
        </div>
        
      </div>
    </app-track-list>
  </div>
</div>
`;
